### PR TITLE
Add billing endpoints for tenant in master api

### PR DESCRIPTION
**Pull request by azgabur**

### CODE REVIEW

File: threescale_api/resources.py [hunk 1]
```diff
@@ -1055,11 +1088,53 @@ def __init__(self, entity_name='name', **kwargs):
 class Tenant(DefaultResource):
     def __init__(self, entity_name='system_name', **kwargs):
         super().__init__(entity_name=entity_name, **kwargs)
+        self._url = self["signup"]["account"]["admin_base_url"]
+        self._token = self["signup"]["access_token"]["value"]
```
[the remark this answers is not in the document]
Comment (azgabur) [reply threescale_api/resources.py]: addressed

File: threescale_api/resources.py [hunk 2]
```diff
@@ -391,6 +393,37 @@ def __init__(self, *args, entity_name='tenant', entity_collection='tenants', **k
     def url(self) -> str:
         return self.threescale_client.master_api_url + '/providers'
 
+    def trigger_billing(self, tenant: Union['Tenant', int], date: str):
```
[the remark this answers is not in the document]
Comment (azgabur) [reply threescale_api/resources.py:396]: addressed

File: threescale_api/resources.py [hunk 3]
```diff
@@ -1055,11 +1088,53 @@ def __init__(self, entity_name='name', **kwargs):
...
+    def wait_tenant_ready(self):
+        return self.tenant_ready
+
+    def api(self, wait=True) -> 'client.ThreeScaleClient':
+        """
+        Returns admin api client for tenant.
+        """
+        if wait:
+            self.wait_tenant_ready()
+        return client.ThreeScaleClient(self.admin_url, self.api_token, ssl_verify=False)
```
[the remark this answers is not in the document]
Comment (azgabur) [reply threescale_api/resources.py]: addressed

File: threescale_api/resources.py [hunk 4]
```diff
@@ -1055,11 +1088,53 @@ def __init__(self, entity_name='name', **kwargs):
...
+        self._url = self["signup"]["account"]["admin_base_url"]
+        self._token = self["signup"]["access_token"]["value"]
 
     @property
     def entity_id(self) -> int:
         return self.entity["signup"]["account"]["id"]
 
+    @property
+    def api_token(self) -> str:
+        return self._token
```
[the remark this answers is not in the document]
Comment (azgabur) [reply threescale_api/resources.py]: addressed

File: threescale_api/resources.py [hunk 5]
```diff
@@ -1055,11 +1088,53 @@ def __init__(self, entity_name='name', **kwargs):
...
+
+    @property
+    def tenant_ready(self) -> bool:
+        """
+        Recommended way of checking if tenant is ready to process api calls.
+        Call in while loop with sleep or use wait_tenant_ready()
+        When True is returned, there is some chance the tenant is actually ready.
+        """
+        return self.api(wait=False).account_plans.exists() and\
+            len(self.api(wait=False).account_plans.list()) >= 1 and\
```
[the remark this answers is not in the document]
Comment (azgabur) [reply threescale_api/resources.py]: addressed

**Review by mganisin — APPROVED** [review]

File: tests/integration/test_integration_tenant.py [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+
+def test_trigger_billing(master_api, custom_tenant):
+    assert master_api.tenants.trigger_billing(custom_tenant, date.today().isoformat())
+
+
+def test_trigger_billing_account(master_api, custom_tenant, account_params):
+    admin_url = custom_tenant["signup"]["account"]["admin_base_url"]
+    token = custom_tenant["signup"]["access_token"]["value"]
+    tenant_api = client.ThreeScaleClient(admin_url, token, ssl_verify=False)
+    wait_for_tenant_initialise(tenant_api)
```
[the remark this answers is not in the document]
Comment (mganisin) [reply tests/integration/test_integration_tenant.py]: @mkudlej this is ready for your review again

**Review by mkudlej — APPROVED** [review]

**Pull request merged**